test_containers/mock_rag_tester/entrypoint.py

```diff
@@ -0,0 +1,119 @@
+
+        # Extract system_under_test
+        sut_params = systems_params.get("system_under_test", {})
+        if not sut_params:
+            raise ValueError("Missing system_under_test in systems_params")
+
+        # Validate SUT type
+        sut_type = sut_params.get("type")
+        if sut_type not in ["rag_api"]:
+            raise ValueError(f"Unsupported system_under_test type: {sut_type}")
+
+        # Extract SUT parameters (flattened structure)
+        base_url = sut_params["base_url"]  # Required, validated upstream
+        api_key = sut_params["api_key"]  # Required, validated upstream
+        model = sut_params["model"]  # Required, validated upstream
+        user_group = sut_params.get("user_group")  # Optional, validated upstream
+
+        extra_body = {} # OpenAI-compatible API extra body parameters
+        if user_group is not None:
+            extra_body = {'user_group': user_group}
+
+        # Extract delay parameter
+        delay_seconds = test_params.get("delay_seconds", 0)
+
+        # Simulate work
+        if delay_seconds > 0:
+            time.sleep(delay_seconds)
+
+        # Simulate RAG API call (this is a mock, so we don't actually call the API)
+        # In a real test container, you would use:
+        # import openai
+        # messages = [{"role": "user", "content": "Hello, world!"}]
+        # client = openai.OpenAI(base_url=base_url, api_key=api_key)
+        # response = client.chat.completions.create(
+        #     model=model,
+        #     messages=messages,
+        #     extra_body=extra_body
+        # )
+
+        # The response variable would contain the actual API response and would need to contain follow the required response format.
+        # content, context = response.choices[0].message.content, response.choices[0].message.context
+
+        # # Validate context format, specific to RAG systems
+        # assert isinstance(context, dict), "Context must be a dictionary"
+        # assert "citations" in context, "Context must contain 'citations' key"
+        # assert isinstance(context["citations"], list), "'citations' must be a list"
+        # for citation in context["citations"]:
+        #     assert isinstance(citation, dict), "Each citation must be a dictionary"
+        #     assert "retrieved_context" in citation, "Citation must contain 'retrieved_context'"
+        #     assert isinstance(citation["retrieved_context"], str), "'retrieved_context' must be a string"
+        #     assert "document_id" in citation, "Citation must contain 'document_id'"
+        #     assert isinstance(citation["document_id"], str), "'document_id' must be a string"
+        #     # Optional fields
+        #     if "score" in citation:
+        #         assert isinstance(citation["score"], (float)), "'score' must be of type float"
+        #     if "source_id" in citation:
+        #         assert isinstance(citation["source_id"], str), "'source_id' must be a string"
```

Comment:
I see what you are doing over here and I think it is a good approach to provide a reusable template for validation, but perhaps we should approach it in a more consistent manner. How about creating at the library level a `rag_response_schema.py` file that others can import and use for validation?

Then the `entrypoint.py` example code could use that function for validation